straxen/corrections_services.py

```diff
@@ -156,17 +155,25 @@ def get_pmt_gains(self, run_id, model_type, global_version,
         to_pe = None
         cache_name = None
 
-        if model_type == 'to_pe_model':
+        if 'to_pe_model' in model_type:
+            # Get the detector name based on the requested model_type
+            # This also will be used to the cachable name convention
+            # pmt == TPC, n_veto == n_veto's PMT, etc
+            detector_names = {'to_pe_model': 'pmt',
+                              'to_pe_model_nv': 'n_veto',
+                              'to_pe_model_mv': 'mu_veto'}
+            target_detector = detector_names[model_type]
+
             if global_version in cacheable_versions:
                 # Try to load from cache, if it does not exist it will be created below
-                cache_name = cacheable_naming(run_id, model_type, global_version)
+                cache_name = cacheable_naming(run_id, model_type, global_version, target_detector)
                 try:
                     to_pe = straxen.get_resource(cache_name, fmt='npy')
                 except (ValueError, FileNotFoundError):
                     pass
 
-            if to_pe is None:
-                to_pe = self._get_correction(run_id, 'pmt', global_version)
+                if to_pe is None:
```

Comment:
Not sure if this is by accident but this is not what you want. Otherwise you cannot load `v1` configs.

Reply:
![afbeelding](https://user-images.githubusercontent.com/22295914/102057064-a0d83700-3ded-11eb-92c4-f25191425de9.png)

Reply:
not sure what do you mean here @jorana, can you elaborate more, please

Reply:
never mind, yes there was an extra indent, good catch!!

Reply:
You over indented this, you should not put it under the `if global_version in cacheable_versions:` if-statement.

If you follow the logic, you want to hit the ` if to_pe is None:` (which you want to do for everything unless you can read it from a file) you cannot do this for e.g. v1 because it is not cacheable.